process/process.go

```diff
@@ -133,6 +201,76 @@ func (p *process) initCapability() error {
 	return nil
 }
 
+// Camouflage() method's purpose is to adjust the process personality to another
+// process. The ultimate goal is to be able to utilize a 'proxy' process (such
+// as 'sysbox-fs nsenter') to execute instructions on behalf of the original
+// process.
+func (p *process) Camouflage(
+	uid uint32,
+	gid uint32,
+	root string,
+	cwd string,
+	caps [2]uint32) error {
+
+	if err := unix.Chdir(cwd); err != nil {
+		return err
+	}
+
+	if err := unix.Chroot(root); err != nil {
+		return err
+	}
+
+	// Initialize capability struct if not already done.
+	if p.cap == nil {
+		if err := p.initCapability(); err != nil {
+			return err
+		}
+	}
+
+	// If UID corresponds to 'root' user then adjust capabilities of this
```

Comment:
Do you mean "non-root" in the container or at host level?

Also, why do we differentiate between root and non-root? Why isn't the setting of the personality identical for both cases?

Reply:
The idea here is to make this method generic enough (beyond our current 'nsenter' use-case) so that it can serve any incoming UID. In the scenario where UID==0 there's no need to execute setresuid/gid.

Putting the use-case aside, i now realize that this "if" statement is missing a "&& gid == 0" condition for it to be correct.